UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/operation/OperationContract.java

```diff
@@ -73,13 +73,17 @@ public String approveTransaction(final Context ctx, final String initiator, fina
         String clientId = cUtil.getEnrollmentIdFromClientId(ctx.getClientIdentity().getId());
         List<String> clientGroups = new GroupContractUtil().getGroupNamesForUser(ctx.getStub(), clientId);
 
-        ApprovalList existingApprovals = operationData.getExistingApprovals().addUsersItem(clientId).addGroupsItems(clientGroups);
+        ApprovalList existingApprovals = operationData.getExistingApprovals();
         ApprovalList requiredApprovals = null;
         try {
             requiredApprovals = AccessManager.getRequiredApprovals(contractName, transactionName, params);
         } catch (MissingTransactionError e) {
             return e.getJsonError();
         }
+        if (requiredApprovals.getUsers().contains(clientId)) {
```

Comment:
We want to add the user itself  and all their groups to the existing approvals if EITHER condition matches.

Reply:
Also, do we need to provide an error, if a user tries to approve a transaction which he has nothing to do with?

Reply:
Yes. Users not participating in a transaction should not be able to see existing approvals for the transaction, as that could leak the proposed transaction parameters.